be/src/olap/rowset/segment_v2/inverted_index/query/query_factory.h

```diff
@@ -0,0 +1,51 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include "olap/rowset/segment_v2/inverted_index_query_type.h"
+
+namespace doris::segment_v2 {
+
+class Query;
+class DisjunctionQuery;
+class ConjunctionQuery;
+class PhraseQuery;
+class PhrasePrefixQuery;
+class RegexpQuery;
+
+class QueryFactory {
+public:
+    template <typename... Args>
+    static std::unique_ptr<Query> create(InvertedIndexQueryType query_type, Args&&... args) {
+        if (query_type == InvertedIndexQueryType::MATCH_ANY_QUERY ||
+            query_type == InvertedIndexQueryType::EQUAL_QUERY) {
+            return std::make_unique<DisjunctionQuery>(std::forward<Args>(args)...);
```

Comment:
maybe switch case is more clear than if else？

Reply:
Change to switch